src/ttrss_api.cpp

```diff
@@ -63,11 +63,15 @@ std::string ttrss_api::retrieve_sid() {
 	std::string sid;
 	try {
 		sid = content["session_id"];
-	} catch (json::exception& e) {}
+	} catch (json::exception& e) {
+		LOG(level::ERROR, "ttrss_api::retrieve_sid: %s", e.what());
+	}
 
 	try {
 		api_level = content["api_level"];
-	} catch (json::exception& e) {};
+	} catch (json::exception& e) {
+		LOG(level::ERROR, "ttrss_api::retrieve_sid: %s", e.what());
```

Comment:
Since the messages are the same, wouldn't it be hard to figure out what exactly went wrong?

Reply:
Just as a side-note, I would object to the use of `level::ERROR` here. The request for `api_level` at this point is a pure optimization for a faster startup. Furthermore for some (older) TTRSS-servers that respond to the `getApiLevel`-request, it is not guaranteed that they will send their `api_version` upon mere authentication. The current code will always (intentionally) fall back to `ttrss_api::query_api_level` and therefore fails silently in the optimized code path. I'm fine with a logmessage, but please be sure to include a more accurate message, e.g.: `Failed to determine api_level during authentication, falling back to explicit detection: %s` and a lower loglevel